tools/PI/DevHome.PI/ViewModels/ExpandedViewControlViewModel.cs

```diff
@@ -236,30 +238,47 @@ public void NavigateTo(Type viewModelType)
             var link = Links[i];
             if (link.PageViewModel == viewModelType)
             {
-                SelectedNavLinkIndex = i;
+                if (SelectedNavLinkIndex == i)
+                {
+                    // Navigation between pages is a bit convoluted, as we set SelectedNavLinkIndex, which triggers a change
+                    // event on a ListView that calls Navigate() to perform the actual navigation.
+                    //
+                    // If we're trying to navigate to the same Listview item, we need to call Navigate() directly. ListViewItems != Pages (for example, Settings
+                    // is a ListView Item, but there are several pages under Settings). If we end up navigating to the same page,
+                    // the NavService will no-op it
+                    Navigate();
```

Comment:
Your instincts are right, that there are dragons here. Once you're in this state...

![image](https://github.com/user-attachments/assets/da7e9f94-9649-4c5e-be60-8a083bff4d1b)

clicking on the Settings item in the ListView on the left doesn't navigate you back to the top level Settings page. Since there isn't a change to the ListView's index, Navigate() doesn't get called. In this case, you'd need to use the breadcrumbs at the top of the page to navigate back to the top level settings page (or click on a different list view item and then click back to Settings).

Reply:
Not sure if this is a bug or not... changing that behavior would involve listening to click events on the ListView instead of the selected index event... I'm hopeful the NavigationView solves all of our problem though. Didn't want to spin too much on it.